### PR TITLE
GQL: `Dataset.visibility()`: return back, after being deleted by mistake

### PR DESCRIPTION
**Pull request by s373r**
### Changed
- Private Datasets:
  - GQL, added `Dataset.visibility()` to get the current visibility value

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,10 +15,10 @@ Recommendation: for ease of reading, use the following order:
 ### Added
 - Console warning when deleting datasets which are out of sync with their push remotes
 ### Changed
-- Speed up project build time by removing unused dependencies which were not detected by automated tools
```
Comment (s373r) [inline CHANGELOG.md:18]: Note for readers: have already backported to the main branch

File: src/adapter/graphql/src/mutations/dataset_mut/dataset_mut.rs [hunk 2]
```diff
@@ -287,24 +288,6 @@ pub enum DatasetVisibilityInput {
     Public(PublicDatasetVisibility),
 }
 
-#[derive(Union, Debug, Clone, PartialEq, Eq)]
-pub enum DatasetVisibility {
```
Comment (s373r) [inline src/adapter/graphql/src/mutations/dataset_mut/dataset_mut.rs:291]: Note for readers: moved to better place

File: src/adapter/graphql/src/queries/datasets/dataset.rs [hunk 3]
```diff
@@ -87,6 +87,26 @@ impl Dataset {
         Ok(summary.kind.into())
     }
 
+    // TODO: Private Datasets: tests
```
Comment (s373r) [inline src/adapter/graphql/src/queries/datasets/dataset.rs:90]: Note for readers: I plan to write tests a little later -- for now, I've limited myself to manual testing

**Review by s373r — COMMENTED** [review]
Added notes for readers

**Review by zaychenko-sergei — APPROVED** [review]

**Pull request merged**